Kudu.Core/Deployment/DeploymentManager.cs

```diff
@@ -693,9 +693,15 @@ private DeployResult GetResult(string id, string activeDeploymentId, bool isDepl
                     {
                         await builder.Build(context);
                         builder.PostBuild(context);
-
-                        await RestartMainSiteIfNeeded(tracer, logger);
-
+                        if(deploymentInfo.Deployer == "OneDeploy")
+                        {
+                            await PostDeploymentHelper.RestartMainSiteAsync(_environment.RequestId, new PostDeploymentTraceListener(tracer, logger));
+                        }
+                        else
+                        {
+                            await RestartMainSiteIfNeeded(tracer, logger);
+                        }
+            
                         await PostDeploymentHelper.SyncFunctionsTriggers(_environment.RequestId, new PostDeploymentTraceListener(tracer, logger), deploymentInfo?.SyncFunctionsTriggersPath);
 
                         if (!_settings.RunFromZip() && _settings.TouchWatchedFileAfterDeployment())
```

Comment:
>TouchWatchedFileAfterDeployment [](start = 65, length = 31)

TouchWatchedFileAfterDeployment() will return true (by default), so TryTouchWatchedFile() will be called. We don't want that as we will be using the RestartMainSiteAsync() to perform a restart.  So to solve that problem, add a new property deploymentInfo.DisableTouchingWatchedFile = false and change the following:
if (!_settings.RunFromZip() && _settings.TouchWatchedFileAfterDeployment())
to:
if (!deploymentInfo.disableTouchingWatchedFile && !_settings.RunFromZip() && _settings.TouchWatchedFileAfterDeployment())

Reply:
if (deploymentInfo.WatchFileEnabled && !_settings.RunFromZip() && _settings.TouchWatchedFileAfterDeployment())

---
In reply to: [456719342](https://github.com/shrishrirang/kudu/pull/2#discussion_r456719342) [](ancestors = 456719342)